packages/editor/src/components/post-switch-to-draft-button/index.js

```diff
@@ -18,37 +22,39 @@ function PostSwitchToDraftButton( {
 	onClick,
 } ) {
 	const isMobileViewport = useViewportMatch( 'small', '<' );
+	const [ showConfirmDialog, setShowConfirmDialog ] = useState( false );
 
 	if ( ! isPublished && ! isScheduled ) {
 		return null;
 	}
 
-	const onSwitch = () => {
-		let alertMessage;
-		if ( isPublished ) {
-			alertMessage = __(
-				'Are you sure you want to unpublish this post?'
-			);
-		} else if ( isScheduled ) {
-			alertMessage = __(
-				'Are you sure you want to unschedule this post?'
-			);
-		}
-		// eslint-disable-next-line no-alert
-		if ( window.confirm( alertMessage ) ) {
-			onClick();
-		}
-	};
+	let alertMessage;
+	if ( isPublished ) {
+		alertMessage = __( 'Are you sure you want to unpublish this post?' );
+	} else if ( isScheduled ) {
+		alertMessage = __( 'Are you sure you want to unschedule this post?' );
+	}
 
 	return (
-		<Button
-			className="editor-post-switch-to-draft"
-			onClick={ onSwitch }
-			disabled={ isSaving }
-			variant="tertiary"
-		>
-			{ isMobileViewport ? __( 'Draft' ) : __( 'Switch to draft' ) }
-		</Button>
+		<>
+			<Button
+				className="editor-post-switch-to-draft"
+				onClick={ () => {
+					setShowConfirmDialog( true );
+				} }
```

Comment:
This is a small optimization, but at the moment the `onClick` callback is being re-created everytime, potentially causing unnecessary re-renders for the `Button` component.

An alternative could be to leverage `useCallback` and memoize that function:

```js
const handleButtonClick = useCallback( () => {
  setShowConfirmDialog( true );
}, [] );

// ...

return (
  <>
    <Button onClick={ handleButtonClick } {/* ... */} />
     {/* ... */} 
  </>
);
```

No dependencies should be needed in the array of dependencies for `useCallback`, since React guarantees that `setShowConfirmDialog`'s identity is stable across re-renders ([#](https://reactjs.org/docs/hooks-reference.html#usestate))